datafusion/expr/src/built_in_function.rs

```diff
@@ -890,8 +890,7 @@ impl BuiltinScalarFunction {
                 // 0 or more arguments of arbitrary type
                 Signature::one_of(vec![VariadicAny, Any(0)], self.volatility())
             }
-            BuiltinScalarFunction::Struct => Signature::variadic(
-                struct_expressions::SUPPORTED_STRUCT_TYPES.to_vec(),
```

Comment:
I think we can also remove `SUPPORTED_STRUCT_TYPES` entirely (I don't think it is used anywhere else)

We can do so as a follow on PR too.